extension.neon

```diff
@@ -289,6 +289,10 @@ services:
         class: Larastan\Larastan\ReturnTypes\CollectionFilterDynamicReturnTypeExtension
         tags:
             - phpstan.broker.dynamicMethodReturnTypeExtension
+    -
+        class: Larastan\Larastan\ReturnTypes\CollectionGroupByReturnTypeExtension
```

Comment:
Is there a way to fix this without having to introduce a new class? I don't think we really want to make a return type extension for every collection method...

Also https://github.com/larastan/larastan/pull/1852 moves the target class from `Collection` to `Enumerable` as this will not work for LazyCollections

Reply:
each collection method has it's own unique workings though which would cause a lot of complexity to manifest in a single place, rather than having a single extension for a single job.

personally i would prefer to see an extension for each method rather than trying to abstract everything into a single extension, it keeps things straightforwards and to the point

Reply:
That's fair, if the logic is different. I guess I would be opposed to two different classes with the same logic